src/types.rs

```diff
@@ -303,34 +325,44 @@ impl Id {
       public_keys : (pub_sign_key, pub_asym_key),
       secret_keys : (sec_sign_key, sec_asym_key),
       validation_token : validation_token,
-      name : NameType::new(digest.0)
+      name : NameType::new(digest.0),
+      original_name : NameType::new(digest.0),
```

Comment:
I'm open for either; but if we don't store original_name, then I would also not store a bool, because at that point it's better to just recalculate the name from the provided fields and compare